### PR TITLE
Reenable background sync on Droid

### LINKED ISSUE #4298: Enable background sync in Android again
We need to undo the changes in #4297 and #4299  and enable background sync in Android again, only after #4230 is solved.

### PR DESCRIPTION
**Pull request by heytherewill**
## What's this?
This re-enables background syn on Android

### Relationships
Closes #4298

## Why do we want this?
Background sync seemed to be responsible for a bunch of crashes and that's why we cut it out. Now we want it back again

## How is it done?
Simply allowing the code that executes the sync to run

### Side effects
The release manager of 2.4 should keep a close watch on crashes and compare them with the crashes that happened in January. OFC we got rid of MVX, so the chances of problems are reduced, but still there.

## Review hints
IIRC @daividssilverio knows how to test this on a device and ensure it's running properly

# :squid: Permissions
Only myself

### CODE REVIEW

File: Toggl.Droid/Services/BackgroundSyncJobSchedulerService.cs [hunk 1]
```diff
@@ -15,11 +15,7 @@ public class BackgroundSyncJobSchedulerService : JobService
 
         public override bool OnStartJob(JobParameters @params)
         {
-            // Background sync is temporary disabled due to a crash that is hard to reproduce
-            // Calling JobFinished and eturning early here stops the background job from running
-            JobFinished(@params, false);
```
Comment (kornelijepetak) [inline Toggl.Droid/Services/BackgroundSyncJobSchedulerService.cs]: Is this line no longer needed?

Comment (heytherewill) [reply Toggl.Droid/Services/BackgroundSyncJobSchedulerService.cs]: No

Comment (kornelijepetak) [reply Toggl.Droid/Services/BackgroundSyncJobSchedulerService.cs]: Man, read your comment 😄 

But yes, I missed the part where the job is being finished after background sync in the subscribe.

**Comment by heytherewill** [comment]
Push notifications can be disabled/not arrive and we'd still have this running, so I say we enable it.

Also, it's up to @maritoggl or @amulware to decide whether we want to run full sync or sync light there

**Comment by amulware** [comment]
Light sync is enough.

**Comment by amulware** [comment]
Obviously be super careful when rolling this out, given the issues we had with it last time!

**Pull request merged**